lib/util/mutex.ts

```diff
@@ -0,0 +1,21 @@
+import { Mutex, MutexInterface, withTimeout } from 'async-mutex';
```

Comment:
```suggestion
import { Mutex, type MutexInterface, withTimeout } from 'async-mutex';
```
isn't that a type only import?